tests/test_modeling_blenderbot.py

```diff
@@ -0,0 +1,467 @@
+
+
+if is_torch_available():
+    import torch
+
+
+@require_torch
+class BlenderbotModelTester:
+    def __init__(
+        self,
+        parent,
+        batch_size=2,
+        seq_len=10,
+        vocab_size=100,
+        hidden_size=16,
+        is_training=False,
+        num_hidden_layers=2,
+        num_attention_heads=2,
+        hidden_dropout_prob=0.2,
+        max_position_embeddings=50,
+        eos_token_id=2,
+        bos_token_id=0,
+        pad_token_id=1,
+        use_labels=True,
+        ffn_size=4,
+        attention_dropout=0.2,
+        activation="gelu",
+        variant="xlm",
+        scale_embedding=True,
+    ):
+        self.parent = parent
+        self.batch_size = batch_size
+        self.seq_len = seq_len
+        self.vocab_size = vocab_size
+        self.hidden_size = hidden_size
+        self.hidden_dropout_prob = hidden_dropout_prob
+        self.is_training = is_training
+        self.num_attention_heads = num_attention_heads
+        self.num_hidden_layers = num_hidden_layers
+        self.bos_token_id = bos_token_id
+        self.eos_token_id = eos_token_id
+        self.pad_token_id = pad_token_id
+        self.max_position_embeddings = max_position_embeddings
+        self.use_labels = use_labels
+        self.ffn_size = ffn_size
+        self.activation = activation
+        self.attention_dropout = attention_dropout
+        self.variant = variant
+        self.scale_embedding = scale_embedding
+        torch.manual_seed(0)
+
+    def prepare_config_and_inputs_for_common(self):
+        input_ids = ids_tensor([self.batch_size, self.seq_len], self.vocab_size)
+
+        config = BlenderbotConfig(
+            d_model=self.hidden_size,
+            dropout=self.hidden_dropout_prob,
+            vocab_size=self.vocab_size,
+            encoder_layers=self.num_hidden_layers,
+            decoder_layers=self.num_hidden_layers,
+            encoder_attention_heads=self.num_attention_heads,
+            decoder_attention_heads=self.num_attention_heads,
+            attention_dropout=self.attention_dropout,
+            encoder_ffn_dim=self.ffn_size,
+            decoder_ffn_dim=self.ffn_size,
+            max_position_embeddings=self.max_position_embeddings,
+            variant=self.variant,
+            scale_embedding=self.scale_embedding,
+            bos_token_id=self.bos_token_id,
+            eos_token_id=self.eos_token_id,
+            pad_token_id=self.pad_token_id,
+            num_beams=1,
+            min_length=3,
+            max_length=10,
+        )
+        attention_mask = ids_tensor([self.batch_size, self.seq_len], vocab_size=2)
+        inputs_dict = {"input_ids": input_ids, "attention_mask": attention_mask}
+        return config, inputs_dict
+
+
+@require_torch
+class BlenderbotTesterMixin(ModelTesterMixin, unittest.TestCase):
+    all_generative_model_classes = (BlenderbotForConditionalGeneration,) if is_torch_available else ()
```

Comment:
We should add the model to common tests here via `all_models = (BlenderbotForConditionalGeneration)`